src/api/app/mixins/webui/has_flags.rb

```diff
@@ -92,4 +92,45 @@ def get_build_flags
     return build_flags
   end
 
+  def existing_flag(options = {})
+    main_object.flags.where(
+      repo: options[:repo], architecture: options[:architecture]
+    ).first
+  end
+
+  def temporary_flag(attributes = {})
+    attributes.update(flag: 'build') # make it configurable
+    attributes[:status] = Flag.default_state('build') unless attributes[:status]
+
+    main_object.flags.new(attributes)
+  end
+
+  def fetch_flag(options = {})
+    existing_flag(options) || temporary_flag(options)
+  end
+
+  def get_build_flags_alt
+    build_flags = {}
+
+    build_flags['all'] = []
+    build_flags['all'] << fetch_flag
+
+    status = build_flags['all'][0].status
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.